### DISCUSSION TITLE
DNS Rewrite *.lan to *.fritz.box (preserve subdomain)

### DISCUSSION THREAD

**Discussion by rdsubhas** [AdguardTeam/AdGuardHome#4532] [Q&A] [answered]
Many of us have routers which have non-configurable LAN domain names. e.g. `fritz.box`

We still use the router's DHCP, because it's helpful with router features (like setting up separate client zones, firewall rules, etc). Let's say we don't want to switch to AGH DHCP just for the lan domain name.

AGH already allows us to set certain TLDs to be resolved by specific upstreams. So pointing `.fritz.box` to the router is not an issue.

Can we rewrite `*.lan` to `*.fritz.box` using AGH, i.e. rewrite the TLD preserving subdomain? It would help all LAN devices internally talk via `<device-name>.lan` domain, but the address resolution can still be done by the upstream router as `<device>.fritz.box`.

**Comment by ainar-g** [comment, marked as answer]
Hello.  That is currently not possible.  You can open an issue and we'll mark it as a feature request.

**Comment by rdsubhas** [comment]
Issue created here for reference, thanks: https://github.com/AdguardTeam/AdGuardHome/issues/4538
